src/client/testing/common/updateTestSettings.ts

```diff
@@ -92,4 +100,49 @@ export class UpdateTestSettingService implements IExtensionActivationService {
             return false;
         }
     }
+
+    private fixLanguageServerSettings(fileContent: string): string {
+        // `python.jediEnabled` is deprecated:
+        //   - `true` or missing then set to `languageServer: Jedi`.
+        //   - `false` and `languageServer` is present, do nothing.
+        //   - `false` and `languageServer` is NOT present, set `languageServer` to `Microsoft`.
+        // `jediEnabled` is then removed.
+        const jediEnabledPath = ['python.jediEnabled'];
+        const languageServerPath = ['python.languageServer'];
+
+        try {
+            let ast = parseTree(fileContent);
+            let jediEnabledNode = findNodeAtLocation(ast, jediEnabledPath);
+            const jediEnabled = jediEnabledNode ? getNodeValue(jediEnabledNode) : true;
+            const languageServerNode = findNodeAtLocation(ast, languageServerPath);
+            const formattingOptions: FormattingOptions = {
+                tabSize: 4,
+                insertSpaces: true
+            };
+            let edits: Edit[] = [];
+
+            if (!jediEnabledNode || jediEnabled) {
+                // `jediEnabled` is missing or is true. Default is true, so assume Jedi.
+                edits = modify(fileContent, languageServerPath, LanguageServerType.Jedi, { formattingOptions });
+            } else {
+                // `jediEnabled` is false. if languageServer is missing, set it to Microsoft.
+                if (!languageServerNode) {
+                    edits = modify(fileContent, languageServerPath, LanguageServerType.Microsoft, {
+                        formattingOptions
+                    });
+                }
+            }
```

Comment:
If `"python.jediEnabled"` is missing then shouldn't it fall back to `"python.languageServer"`?

```suggestion
            if (jediEnabledNode) {
                // We overwrite any existing "python.languageServer" setting
                // since it was already ignored anyway.
                value = getNodeValue(jediEnabledNode)
                    ? LanguageServerType.Jedi
                    : LanguageServerType.Microsoft;
                edits = modify(fileContent, languageServerPath, value, { formattingOptions });
                if (languageServerNode && getNodeValue(languageServerNode) !== value) {
                    // Comment out the old value?
                }
            } else if (!languageServerNode) {
                // `jediEnabled` is missing.  So is `languageServer` so set it to the default.
                const value = this.experiments.inExperiment(LSEnabled)
                    ? LanguageServerType.Microsoft
                    : LanguageServerType.Jedi;
                edits = modify(fileContent, languageServerPath, value, {
                    formattingOptions
                });
            }
```

Reply:
No. `python.jediEnabled` missing means `true` since this is default in `project.json`.

Reply:
I thought the current behavior is that we fall back to `"python.languageServer"` if `"python.jediEnabled'" isn't set.

Reply:
Ah, never mind.  VS Code applies that default which we then read.  So at what point was `"python.languageServer"` actually used prior to this PR?

Reply:
If `jediEnabled` is set to false, then this setting was consulted (defaulted to Microsoft).

Reply:
History is that there was no `languageServer`, just `jediEnabled`. `jediEnabled == false` meant use Microsoft LS. Then some folks wanted `None` in addition to default so `languageServer` was born.